common/djangoapps/third_party_auth/signals/handlers.py

```diff
@@ -0,0 +1,61 @@
+"""
+Signal handlers for third_party_auth app.
+"""
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from edx_django_utils.monitoring import set_custom_attribute
+
+
+from common.djangoapps.third_party_auth.models import SAMLConfiguration, SAMLProviderConfig
+
+
+from common.djangoapps.third_party_auth.toggles import ENABLE_SAML_CONFIG_SIGNAL_HANDLERS
+
+
+@receiver(post_save, sender=SAMLConfiguration)
+def update_saml_provider_configs_on_configuration_change(sender, instance, created, **kwargs):
+    """
+    NOTE: This behavior is controlled by the ENABLE_SAML_CONFIG_SIGNAL_HANDLERS toggle.
+
+    Signal handler to create a new SAMLProviderConfig when SAMLConfiguration is updated.
+
+    When a SAMLConfiguration is updated and a new version is created,
+    this handler generates a corresponding
+    SAMLProviderConfig that references the latest configuration version,
+    ensuring all providers remain aligned
+    with the most current settings.
+    """
+    # .. custom_attribute_name: saml_config_signal.enabled
+    # .. custom_attribute_description: Tracks whether the SAML config signal handler is enabled.
+    set_custom_attribute('saml_config_signal.enabled', ENABLE_SAML_CONFIG_SIGNAL_HANDLERS.is_enabled())
+
+    # .. custom_attribute_name: saml_config_signal.new_config_id
+    # .. custom_attribute_description: Records the ID of the new SAML configuration instance.
+    set_custom_attribute('saml_config_signal.new_config_id', instance.id)
+
+    # .. custom_attribute_name: saml_config_signal.slug
+    # .. custom_attribute_description: Records the slug of the SAML configuration instance.
+    set_custom_attribute('saml_config_signal.slug', instance.slug)
+
+    if ENABLE_SAML_CONFIG_SIGNAL_HANDLERS.is_enabled():
+        try:
+            # Find all EXISTING SAMLProviderConfig instances (current_set) that should be
```

Comment:
Nit: Not sure why we need all-caps.
```suggestion
            # Find all existing SAMLProviderConfig instances (current_set) that should be
```